src/mmda/recipes/recipe.py

```diff
@@ -0,0 +1,62 @@
+"""
+
+
+@kylel
+
+"""
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+from mmda.types import *
+from mmda.parsers.pdfplumber_parser import PDFPlumberParser
+from mmda.rasterizers.rasterizer import PDF2ImageRasterizer
+from mmda.predictors.heuristic_predictors.dictionary_word_predictor import DictionaryWordPredictor
+from mmda.predictors.lp_predictors import LayoutParserPredictor
+from mmda.predictors.hf_predictors.token_classification_predictor import \
+    IVILATokenClassificationPredictor
+
+
+class Recipe:
```

Comment:
Can we move this abstract class in a separate file (call it `base.py`) and rename it to `AbstractRecipe`? So it's clear you should not use it directly.

Reply:
i went with `from abc import abstractmethod` route if thats ok

Reply:
also separated into separate files